### PR TITLE
Add tests to the TLS certificate reload

### PR DESCRIPTION
**Pull request by cescoffier**
This PR uses the new certificate generator utility.

Once in, we will be able to remove all the hand-crafted certificates.

### CODE REVIEW

**Comment by cescoffier** [comment]
Ok... I broke everything :-). Will have a look Monday.

**Comment by sberyozkin** [comment]
@cescoffier Sorry, also missed the review request,  it looks promising in any case 👍

**Comment by cescoffier** [comment]
@sberyozkin Fixed! Your review would be much appreciated. 

The goal is to use this technique to avoid hand-crafted certificates. Here, it's slightly more complicated as I'm testing the hot reload, which means I need two sets of certificates: the initial certificate and the one triggering the reload.

In the other case, we should have only one.

Also, while adding that feature to the REST client, we discovered that the REST client did not support PEMs and P12 (getting fixed right now), and the trust store format needs to be completed. It is the case with gRPC, too. Because creating certificates was painful, I can understand why it was not adequately tested. So, a consequence would be the addition of more certificate formats in various extensions.

**Review by geoand — APPROVED** [review]
Nice!

File: bom/application/pom.xml [hunk 1]
```diff
@@ -3593,6 +3593,11 @@
                 <artifactId>hamcrest</artifactId>
                 <version>${hamcrest.version}</version>
             </dependency>
+            <dependency>
+                <groupId>me.escoffier.certs</groupId>
+                <artifactId>certificate-generator-junit5</artifactId>
+                <version>0.3.0</version>
+            </dependency>
```
Comment (gsmet) [inline bom/application/pom.xml:3600]: @cescoffier I don't think this should be in the BOM. I would move it to the `build-parent`.

Comment (cescoffier) [reply bom/application/pom.xml:3600]: Ok, I will do that in my next PR about kafka TLS support (which as I needed to fix something in the junit5 extension)

**Pull request merged**